### DISCUSSION TITLE
Reopen : Is it possible to use third party library in worklet? #3330

### DISCUSSION THREAD

**Discussion by sumitmanral** [software-mansion/react-native-reanimated#5389] [Q&A] [answered]
I am using GestureDetectors and have a grid and  l have all cell draw functions as worklets .I have used value formatters for the text, number , date etc which call 3 rd party libs like numbro , date-fns . Is there a way I can use runOnJS and wait on the result . Please point me to an example where we are awaiting on values set on runOnJS .

**Comment by tomekzaw** [comment]
Unfortunately, it is not possible to wait for the result in worklets. Currently you cannot use functions from `numbro` or `date-fns` because they are not workletized. You can either implement them on your own as worklets or use built-in `Intl` module.

**Reply by fax1ty** [reply]
How can I make it with Reanimated 3? 
https://github.com/wcandillon/can-it-be-done-in-react-native/blob/master/season5/src/Headspace/Play.tsx#L16

**Reply by MatiPl01** [reply]
You would have to copy-paste code and add `"worklet"` string at the beginning of each function you want to call on the UI thread (inside a worklet).

**Reply by fax1ty** [reply]
Is it ok to call third-party modules inside of a worklet? 
![image](https://github.com/user-attachments/assets/93b9746c-e81f-4691-97b2-4eb12703b623)

**Reply by MatiPl01** [reply, marked as answer]
I can see your issue now. Probably `MakeFromSVGString` is not a worklet function, so you won't be able to create a Skia Path inside a worklet. The `MakeFromSVGString` method should be workletized in the `react-native-skia` library in order to make it work on the UI thread.

Usually, you won't be able to call third-party modules inside worklets because they do not provide workletized versions of exported functions/methods.
